packages/form-field/src/FormField/FormField.tsx

```diff
@@ -0,0 +1,121 @@
+  inputWrapperDisabledStyles,
+  inputWrapperFocusStyles,
+  inputWrapperModeStyles,
+  inputWrapperSizeStyles,
+  inputWrapperStateStyles,
+  textContainerStyle,
+} from './FormField.styles';
+import { FormFieldProps, FormFieldState } from './FormField.types';
+
+export const FormField = forwardRef<HTMLDivElement, FormFieldProps>(
+  (
+    {
+      label,
+      description,
+      children,
+      state = FormFieldState.Unset,
+      size,
+      disabled,
+      notificationMessage,
+      icon,
+      className,
+      inputWrapperProps,
+      ...rest
+    }: FormFieldProps,
+    fwdRef,
+  ) => {
+    const { theme } = useDarkMode();
+    const baseFontSize = useUpdatedBaseFontSize();
+
+    const labelId = useIdAllocator({ prefix: 'lg-form-field-label' });
+    const descriptionId = useIdAllocator({
+      prefix: 'lg-form-field-description',
+    });
+    const errorId = useIdAllocator({ prefix: 'lg-form-field-description' });
+    const inputId = useIdAllocator({ prefix: 'lg-form-field-input' });
+
+    return (
+      <div
+        className={cx(formFieldFontStyles[baseFontSize], className)}
+        ref={fwdRef}
+        {...rest}
+      >
+        <div className={textContainerStyle}>
+          {label && (
+            <Label htmlFor={inputId} id={labelId} disabled={disabled}>
+              {label}
+            </Label>
+          )}
+          {description && (
+            <Description id={descriptionId} disabled={disabled}>
+              {description}
+            </Description>
+          )}
+        </div>
+        {/* eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions */}
+        <div
+          {...inputWrapperProps}
+          aria-disabled={disabled}
+          tabIndex={-1}
+          className={cx(
+            inputWrapperBaseStyles,
+            inputWrapperModeStyles[theme],
+            inputWrapperFocusStyles[theme],
+            inputWrapperSizeStyles[size ?? Size.Default],
+            inputWrapperStateStyles[state][theme],
+            {
+              [inputWrapperDisabledStyles[theme]]: disabled,
+            },
+            inputWrapperProps?.className,
+          )}
+        >
+          <div className={childrenWrapperStyles}>
+            {React.cloneElement(children, {
+              id: inputId,
+              'aria-labelledby': labelId,
+              'aria-describedby': `${descriptionId} ${errorId}`,
+            })}
+          </div>
+          {state === FormFieldState.Error && (
+            <Icon glyph="Warning" className={errorIconStyles[theme]} />
+          )}
+          {icon &&
+            React.cloneElement(icon, {
+              className: cx(
+                iconClassName,
+                iconStyles[theme],
+                icon.props.className,
+              ),
+              disabled,
+            })}
+        </div>
+        {state === FormFieldState.Error && (
+          <Error id={errorId}>{notificationMessage}</Error>
```

Comment:
this should def be errorMessage since it's only showing up when the state is State.Error

Reply:
Password shows a message when the state is "valid". Should these be separate strings?

Reply:
yes bc we're only surfacing this when "error" that might be an edge case that's not captured in the initial API